ziti/cmd/ziti/cmd/create_config.go

```diff
@@ -18,49 +18,169 @@ package cmd
+)
+
+// CreateConfigOptions the options for the create config command
 type CreateConfigOptions struct {
 	common.CommonOptions
-	Namespace   string
-	Version     string
-	ReleaseName string
-	HelmUpdate  bool
+
+	Output       string
+	DatabaseFile string
 }
 
-// NewCmdCreateConfig creates a command object for the "create" command
-func NewCmdCreateConfig(p common.OptionsProvider) *cobra.Command {
-	options := &CreateConfigOptions{
-		CommonOptions: p(),
-	}
+type ConfigTemplateValues struct {
+	ZitiPKI                  string
+	ZitiCtrlIntermediateName string
+	ZitiCtrlHostname         string
+	ZitiFabCtrlPort          string
+
+	// Controller specific
+	CtrlListener                 string
+	MgmtListener                 string
+	ZitiHome                     string
+	Hostname                     string
+	ZitiFabMgmtPort              string
+	ZitiEdgeCtrlAPI              string
+	ZitiSigningIntermediateName  string
+	ZitiEdgeCtrlPort             string
+	ZitiCtrlRawname              string
+	ZitiEdgeCtrlIntermediateName string
+	ZitiEdgeCtrlHostname         string
+
+	// Router specific
+	RouterName             string
+	ZitiEdgeRouterHostname string
+	ZitiEdgeRouterPort     string
+	ZitiEdgeWSSRouterName  string
+	WssEnabled             bool
+	IsPrivate              bool
+	IsFabricRouter         bool
+}
 
+// NewCmdCreateConfig creates a command object for the "config" command
+func NewCmdCreateConfig(f cmdutil.Factory, out io.Writer, errOut io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:     "config",
 		Short:   "Creates a config file for specified Ziti component",
 		Aliases: []string{"cfg"},
 		Run: func(cmd *cobra.Command, args []string) {
-			options.Cmd = cmd
-			options.Args = args
-			err := options.Run()
-			cmdhelper.CheckErr(err)
+			cmdhelper.CheckErr(cmd.Help())
 		},
 	}
 
-	cmd.AddCommand(NewCmdCreateConfigController(p))
-	cmd.AddCommand(NewCmdCreateConfigRouter(p))
+	// Get env variable data global to all config files
+	templateData := &ConfigTemplateValues{}
+	templateData.populateEnvVars()
+
+	cmd.AddCommand(NewCmdCreateConfigController(templateData))
+	cmd.AddCommand(NewCmdCreateEnvironment(f, out, errOut))
+	cmd.AddCommand(NewCmdCreateConfigRouter(templateData))
 
-	options.addFlags(cmd, "", "")
 	return cmd
 }
 
 // Add flags that are global to all "create config" commands
-func (options *CreateConfigOptions) addFlags(cmd *cobra.Command, defaultNamespace string, defaultOptionRelease string) {
+func (options *CreateConfigOptions) addCreateFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().BoolVarP(&options.Verbose, optionVerbose, "v", defaultVerbose, verboseDescription)
+	cmd.PersistentFlags().StringVarP(&options.Output, optionOutput, "o", defaultOutput, outputDescription)
 }
 
-// Run implements this command
-func (options *CreateConfigOptions) Run() error {
-	return options.Cmd.Help()
+func (data *ConfigTemplateValues) populateEnvVars() {
```

Comment:
was the code in cmd.go not 'working' properly that you need to do this for the "ziti" vars? or does it not get executed?

specifically this bit of code:
```
	viper.SetEnvPrefix(c.ZITI) // All env vars we seek will be prefixed with "ZITI_"
	viper.AutomaticEnv()
	replacer := strings.NewReplacer("-", "_") // We use underscores in env var names, but use dashes in flag names
	viper.SetEnvKeyReplacer(replacer)
```
does that not 'work' for our needs here?